Makefile

```diff
@@ -4,37 +4,35 @@ SHELL = /bin/bash
 
 CURRENT_DIR := $(shell pwd)
 INSTALL_DIR := $(CURRENT_DIR)/.venv
+PYTHON_LOCAL_DIR := $(CURRENT_DIR)/build/local
 PYTHON_FILES := $(shell find ./* -type f -name "*.py" -print)
 
 #FIXME: put this variable in config file
-PYTHON_VERSION=3.7.4
-CURRENT_PYTHON_VERSION := $(shell python3 -c "import sys;t='{v[0]}.{v[1]}.{v[2]}'.format(v=list(sys.version_info[:]));sys.stdout.write(t)")
-PYTHON_VERSION_OK := $(shell python3 -c "import sys; t=sys.version_info[0:3]; print(int('{}.{}.{}'.format(*t) == '$(PYTHON_VERSION)'))")
+PYTHON_VERSION := 3.7.4
+SYSTEM_PYTHON_CMD := $(shell ./getPythonCmd.sh ${PYTHON_VERSION} ${PYTHON_LOCAL_DIR})
 
 # Commands
-SYSTEM_PYTHON_CMD := $(shell which python3)
 PYTHON_CMD := $(INSTALL_DIR)/bin/python3
 PIP_CMD := $(INSTALL_DIR)/bin/pip3
 FLASK_CMD := $(INSTALL_DIR)/bin/flask
 YAPF_CMD := $(INSTALL_DIR)/bin/yapf
+PYLINT_CMD := $(INSTALL_DIR)/bin/pylint
 all: help
 
 # This bit check define the build/python "target": if the system has an acceptable version of python, there will be no need to install python locally.
 
-ifeq ($(PYTHON_VERSION_OK),1)
-PYTHON_BINDIR := $(shell dirname $(PYTHON_CMD))
-PYTHONHOME := $(shell eval "cd $(PYTHON_BINDIR); pwd; cd > /dev/null")
+ifneq ($(SYSTEM_PYTHON_CMD),)
 build/python:
-		@echo $(CURRENT_PYTHON_VERSION)
-		@echo $(shell $(PYTHON_CMD) -c "print('OK')")
-		mkdir -p build
-		touch build/python;
-else
-build/python: local/bin/python3.7
+	@echo "Using system" $(shell $(SYSTEM_PYTHON_CMD) --version 2>&1)
+	@echo $(shell $(SYSTEM_PYTHON_CMD) -c "print('OK')")
 	mkdir -p build
-	touch build/python;
+	touch build/python
+else
+build/python: $(PYTHON_LOCAL_DIR)/bin/python3.7
```

Comment:
If you have a `PYTHON_VERSION` you should also use a variable for `python3.7`, in the case we use another version of python

Reply:
Good point I'll make the change in another pull-request as it doesn't have anything to do with this PR